doppler/resource_secret.go

```diff
@@ -75,11 +84,13 @@ func resourceSecretUpdate(ctx context.Context, d *schema.ResourceData, m interfa
 	name := d.Get("name").(string)
 	value := d.Get("value").(string)
 	visibility := d.Get("visibility").(string)
+	valueType := d.Get("value_type").(string)
```

Comment:
In addition to adding this field to the `resourceSecretUpdate` func, we need to add it to `resourceSecretRead` to detect out-of-band changes to the secret value type. Unfortunately, the [GET /v3/configs/config/secret](https://docs.doppler.com/reference/secrets-get) endpoint that we use today doesn't have this data available. We can discuss in ENG-8693 how to go about adding it